[Windows] SDL2 archives are sometimes unzipped to the wrong directory

When unzipping the downloaded SDL2 binaries, the `cmake -E tar` process is run 
using whatever current working directory cmake is using; when opening the 
project in Visual Studio directly, that happens to be the build directory.

If you use the CMake GUI, it could be `C:\Program Files\CMake\bin`;
if you do something like `cmake -B build -S .` on the command-line (instead of the 
traditional `mkdir build && cd build && cmake ..`) it will be the source directory.

As a result it may either fail to unzip:
```
Downloading SDL2 Development Library...
CMake Error: Problem with archive_write_header(): Can't create '\\?\C:\Program Files\CMake\bin\SDL2-2.0.14'
```

Or, if it does unzip, then find_package fails to find it:
```
-- Downloading SDL2 Development Library...
-- Could NOT find SDL2 (missing: SDL2_INCLUDE_DIR SDL2_LIBRARIES) (Required is at least version "2.0.7")
```

Adding `WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"` to those execute_process 
would make it always unzip to the build directory.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -142,14 +142,16 @@ if(NOT SDL2_DIR)
                 "${CMAKE_CURRENT_BINARY_DIR}/SDL2-VC.zip"
                 EXPECTED_HASH SHA256=232071cf7d40546cde9daeddd0ec30e8a13254c3431be1f60e1cdab35a968824)
             execute_process(COMMAND "${CMAKE_COMMAND}" -E tar xf
-                "${CMAKE_CURRENT_BINARY_DIR}/SDL2-VC.zip")
+                "${CMAKE_CURRENT_BINARY_DIR}/SDL2-VC.zip"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
         else()
             file(DOWNLOAD
                 "https://www.libsdl.org/release/SDL2-devel-2.0.14-mingw.tar.gz"
                 "${CMAKE_CURRENT_BINARY_DIR}/SDL2-mingw.tar.gz"
                 EXPECTED_HASH SHA256=405eaff3eb18f2e08fe669ef9e63bc9a8710b7d343756f238619761e9b60407d)
             execute_process(COMMAND "${CMAKE_COMMAND}" -E tar xf
-                "${CMAKE_CURRENT_BINARY_DIR}/SDL2-mingw.tar.gz")
+                "${CMAKE_CURRENT_BINARY_DIR}/SDL2-mingw.tar.gz"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
         endif()
         set(SDL2_DIR "${CMAKE_CURRENT_BINARY_DIR}/SDL2-2.0.14")
     endif()
@@ -166,14 +168,16 @@ if(NOT SDL2_IMAGE_DIR)
                 "${CMAKE_CURRENT_BINARY_DIR}/SDL2_image-VC.zip"
                 EXPECTED_HASH SHA256=a180f9b75c4d3fbafe02af42c42463cc7bc488e763cfd1ec2ffb75678b4387ac)
             execute_process(COMMAND "${CMAKE_COMMAND}" -E tar xf
-                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_image-VC.zip")
+                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_image-VC.zip"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
         else()
             file(DOWNLOAD
                 "https://www.libsdl.org/projects/SDL_image/release/SDL2_image-devel-2.0.5-mingw.tar.gz"
                 "${CMAKE_CURRENT_BINARY_DIR}/SDL2_image-mingw.tar.gz"
                 EXPECTED_HASH SHA256=41d9e5ff98aa84cf66e6c63c78e7c346746982fa53d3f36633423cc9177f986c)
             execute_process(COMMAND "${CMAKE_COMMAND}" -E tar xf
-                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_image-mingw.tar.gz")
+                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_image-mingw.tar.gz"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
         endif()
         set(SDL2_IMAGE_DIR "${CMAKE_CURRENT_BINARY_DIR}/SDL2_image-2.0.5")
     endif()
@@ -190,14 +194,16 @@ if(NOT SDL2_MIXER_DIR)
                 "${CMAKE_CURRENT_BINARY_DIR}/SDL2_mixer-VC.zip"
                 EXPECTED_HASH SHA256=258788438b7e0c8abb386de01d1d77efe79287d9967ec92fbb3f89175120f0b0)
             execute_process(COMMAND "${CMAKE_COMMAND}" -E tar xf
-                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_mixer-VC.zip")
+                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_mixer-VC.zip"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
         else()
             file(DOWNLOAD
                 "https://www.libsdl.org/projects/SDL_mixer/release/SDL2_mixer-devel-2.0.4-mingw.tar.gz"
                 "${CMAKE_CURRENT_BINARY_DIR}/SDL2_mixer-mingw.tar.gz"
                 EXPECTED_HASH SHA256=14250b2ade20866c7b17cf1a5a5e2c6f3920c443fa3744f45658c8af405c09f1)
             execute_process(COMMAND "${CMAKE_COMMAND}" -E tar xf
-                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_mixer-mingw.tar.gz")
+                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_mixer-mingw.tar.gz"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
         endif()
         set(SDL2_MIXER_DIR "${CMAKE_CURRENT_BINARY_DIR}/SDL2_mixer-2.0.4")
     endif()
@@ -214,14 +220,16 @@ if(NOT SDL2_NET_DIR)
                 "${CMAKE_CURRENT_BINARY_DIR}/SDL2_net-VC.zip"
                 EXPECTED_HASH SHA256=c1e423f2068adc6ff1070fa3d6a7886700200538b78fd5adc36903a5311a243e)
             execute_process(COMMAND "${CMAKE_COMMAND}" -E tar xf
-                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_net-VC.zip")
+                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_net-VC.zip"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
         else()
             file(DOWNLOAD
                 "https://www.libsdl.org/projects/SDL_net/release/SDL2_net-devel-2.0.1-mingw.tar.gz"
                 "${CMAKE_CURRENT_BINARY_DIR}/SDL2_net-mingw.tar.gz"
                 EXPECTED_HASH SHA256=fe0652ab1bdbeae277d7550f2ed686a37a5752f7a624f54f19cf1bd6ba5cb9ff)
             execute_process(COMMAND "${CMAKE_COMMAND}" -E tar xf
-                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_net-mingw.tar.gz")
+                "${CMAKE_CURRENT_BINARY_DIR}/SDL2_net-mingw.tar.gz"
+                WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
         endif()
         set(SDL2_NET_DIR "${CMAKE_CURRENT_BINARY_DIR}/SDL2_net-2.0.1")
     endif()

